apps/ledger-live-mobile/src/screens/Swap/LiveApp/hooks/useSwapLiveAppCustomHandlers.ts

```diff
@@ -1,30 +1,16 @@
 import { LiveAppManifest } from "@ledgerhq/live-common/platform/types";
 import { WalletAPICustomHandlers } from "@ledgerhq/live-common/wallet-api/types";
-import { useNavigation } from "@react-navigation/native";
 import { useMemo } from "react";
 import { useDispatch, useSelector } from "react-redux";
-import { usePTXCustomHandlers } from "~/components/WebPTXPlayer/CustomHandlers";
 import { flattenAccountsSelector } from "~/reducers/accounts";
-import { swapCustomHandlers } from "../customHandlers";
+import { useSwapCustomHandlers } from "../customHandlers";
 
 export function useSwapLiveAppCustomHandlers(manifest: LiveAppManifest) {
   const accounts = useSelector(flattenAccountsSelector);
-  const ptxCustomHandlers = usePTXCustomHandlers(manifest, accounts);
   const dispatch = useDispatch();
-  const navigation = useNavigation();
+  const swapCustomHandlers = useSwapCustomHandlers(manifest, accounts, dispatch);
 
   return useMemo<WalletAPICustomHandlers>(() => {
-    // Cast to unknown first to handle type incompatibilities
-    const customHandlers = {
-      ...ptxCustomHandlers,
-      ...swapCustomHandlers({
-        accounts,
-        dispatch,
-        navigation,
-      }),
-    } as unknown;
-
-    // Then cast to the final type
-    return customHandlers as WalletAPICustomHandlers;
-  }, [ptxCustomHandlers, accounts, dispatch, navigation]);
+    return swapCustomHandlers as WalletAPICustomHandlers;
```

Comment:
If we know the type it should be possible to return to correctly typed object from the hook without having to type cast here, no?!